### PR TITLE
API test for managing secrets privilege

### PR DESCRIPTION
**Pull request by kkorchak**
API test for managing secrets privilege

Test steps:
1. Disable 'All users' privileges
2. Create a new user  
3. Verify new user can't create secrets
4. Assign privileges to the new user to manage secrets
5. Verify new user can create and manage secrets
6. Remove the policy
7. Ensure the user can't create a secret after the policy is removed
8. Remove test user
9. Restore All users privileges

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: smoke-test/tests/privileges/test_manage_secrets.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    yield get_frontend_session()
+
+
+@pytest.mark.dependency(depends=["test_healthchecks"])
+@pytest.fixture(scope="module", autouse=True)
+def privileges_and_test_user_setup(admin_session):
+    """Fixture to execute setup before and tear down after all tests are run"""
+    # Disable 'All users' privileges
+    base_privileges_set_status("INACTIVE", admin_session)
+    sensitive_info_prifileges_set_status("INACTIVE", admin_session)
```
Comment (jjoyce0510) [inline smoke-test/tests/privileges/test_manage_secrets.py:34]: typo here!

Comment (jjoyce0510) [reply smoke-test/tests/privileges/test_manage_secrets.py:34]: what are sensitive_info privileges?

File: smoke-test/tests/privileges/test_manage_secrets.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    view_entity_prifileges_set_status("INACTIVE", admin_session)
```
Comment (jjoyce0510) [inline smoke-test/tests/privileges/test_manage_secrets.py:35]: typo here!

File: smoke-test/tests/privileges/utils.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
+import requests_wrapper as requests
+from tests.consistency_utils import wait_for_writes_to_sync
+from tests.utils import (get_frontend_url, wait_for_writes_to_sync, get_admin_credentials)
+
+
+def base_privileges_set_status(status, session):
```
Comment (jjoyce0510) [inline smoke-test/tests/privileges/utils.py:6]: let's name this with action first:

set_base_platform_privileges_status

File: smoke-test/tests/privileges/utils.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            },
+        },
+    }
+    sensitive_info_response = session.post(
+        f"{get_frontend_url()}/api/v2/graphql", json=dataset_sensitive_information)
+    sensitive_info_response.raise_for_status()
+    sens_info_data = sensitive_info_response.json()
+    assert sens_info_data["data"]["updatePolicy"] == "urn:li:dataHubPolicy:view-dataset-sensitive"
+
+def view_entity_prifileges_set_status(status, session):
```
Comment (jjoyce0510) [inline smoke-test/tests/privileges/utils.py:75]: set_view_entity_profile_privileges_policy_status

**Review by jjoyce0510 — COMMENTED** [review]
Overall this is looking really really good! Left a few comments.

File: smoke-test/tests/privileges/test_manage_secrets.py (hunk 1 above)
Comment (kkorchak) [reply smoke-test/tests/privileges/test_manage_secrets.py:34]: @jjoyce0510 
image

**Comment by jjoyce0510** [comment]
Nice looking really good. Let's make sure that these new tests pass!

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**